qiskit_experiments/framework/package_deps.py

```diff
@@ -10,12 +10,62 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 """
-Functions for checking dependency versions.
+Functions for checking and reporting installed package versions.
 """
 
-from importlib.metadata import version
+from __future__ import annotations
 
+from functools import lru_cache
+from importlib.metadata import version as metadata_version
 
-def numpy_version():
-    """Returns the current numpy version in (major, minor) form."""
-    return tuple(map(int, version("numpy").split(".")[:2]))
+from packaging.version import Version
+
+from qiskit.utils.lazy_tester import LazyImportTester
+
+
+__all__ = ["HAS_SKLEARN", "HAS_DYNAMICS", "qiskit_version", "version_is_at_least"]
+
+
+HAS_SKLEARN = LazyImportTester(
+    {
+        "sklearn.discriminant_analysis": (
+            "LinearDiscriminantAnalysis",
+            "QuadraticDiscriminantAnalysis",
+        )
+    },
+    name="scikit-learn",
+    install="pip install scikit-learn",
+)
+
+HAS_DYNAMICS = LazyImportTester(
+    "qiskit_dynamics",
+    name="qiskit-dynamics",
+    install="pip install qiskit-dynamics",
+)
+
+
+def qiskit_version() -> dict[str, str]:
+    """Return a deict with Qiskit names and versions."""
+    return {p: metadata_version(p) for p in ("qiskit", "qiskit-experiments")}
+
+
+@lru_cache(maxsize=None)
+def version_is_at_least(package: str, version: str) -> bool:
+    """Return True if the installed version of package greater than minimum version
+
+    Args:
+        package: Name of the package
+        version: Minimum version name as a string. This should just include
+            major, minor, and micro parts. The function will add ``.dev0`` to
+            also catch any pre-release versions (otherwise ``0.5.0a1`` would
+            evaluate as less than ``0.5.0``).
+
+    Returns:
+        True if installed version greater than ``version``. False if it is less.
+
+    Raises:
+        PackageNotFoundError:
+            If ``package`` is not installed.
+    """
+    installed_version = Version(metadata_version(package))
+    return installed_version > Version(f"{version}.dev0")
```

Comment:
This is nice! It seems like it can't handle some [unusual version strings](https://news.ycombinator.com/item?id=36986538) but I don't think we would encounter any of them. Perhaps we could catch any potential errors here just to be safe.

Reply:
One idea -- PEP440 is pretty old. Anything we are likely to care about should be following it by now. So if we get a version that doesn't parse, we can assume it is old and return False.

Reply:
I like that! And emit a warning about the version string.

Reply:
Let me know if the new commit looks good.

Reply:
The new code looks good. Should the greater than sign be greater than and equal to though? Right now `0.5.0.dev1` is considered to be at least `0.5.0` but `0.5.0.dev0` is not.

Reply:
Ah, good, point. I will fix that and add the PR to the merge queue unless you notice something else :slightly_smiling_face: